### PR TITLE
JS-261 Update easy deps

### PR DESCRIPTION
**Pull request by zglicz**
Use `ncu -i --format group` and for now, only bump the easy ones.

### CODE REVIEW

File: sonar-plugin/sonar-javascript-plugin/pom.xml [hunk 1]
```diff
@@ -363,14 +363,14 @@
...
                     <file>${project.build.directory}/${project.build.finalName}-multi.jar</file>
                   </files>
                 </requireFilesSize>
                 <requireFilesSize>
                   <minsize>30000000</minsize>
-                  <maxsize>50000000</maxsize>
+                  <maxsize>51000000</maxsize>
                   <files>
                     <file>${project.build.directory}/${project.build.finalName}-linux-x64.jar</file>
                     <file>${project.build.directory}/${project.build.finalName}-win-x64.jar</file>
```
Comment (zglicz) [inline sonar-plugin/sonar-javascript-plugin/pom.xml:376]: Updating the sizes, as the jar is just slightly above the limits, so it seems like only a slight increase.

`
[ERROR] Rule 0: org.apache.maven.enforcer.rules.files.RequireFilesSize failed with message:
[ERROR] /tmp/cirrus-ci-build/sonar-plugin/sonar-javascript-plugin/target/sonar-javascript-plugin-10.15.0-SNAPSHOT-multi.jar size (87528440) too large. Max. is 87500000/tmp/cirrus-ci-build/sonar-plugin/sonar-javascript-plugin/target/sonar-javascript-plugin-10.15.0-SNAPSHOT-multi.jar`
`[ERROR] Rule 1: org.apache.maven.enforcer.rules.files.RequireFilesSize failed with message:
[ERROR] /tmp/cirrus-ci-build/sonar-plugin/sonar-javascript-plugin/target/sonar-javascript-plugin-10.15.0-SNAPSHOT-linux-x64.jar size (50579040) too large. Max. is 50000000/tmp/cirrus-ci-build/sonar-plugin/sonar-javascript-plugin/target/sonar-javascript-plugin-10.15.0-SNAPSHOT-linux-x64.jar
[ERROR]`
`[ERROR] /tmp/cirrus-ci-build/sonar-plugin/sonar-javascript-plugin/target/sonar-javascript-plugin-10.15.0-SNAPSHOT.jar size (27659105) too large. Max. is 27500000/tmp/cirrus-ci-build/sonar-plugin/sonar-javascript-plugin/target/sonar-javascript-plugin-10.15.0-SNAPSHOT.jar`

**Review by vdiez — CHANGES_REQUESTED** [review]
Please sync versions with `packages/jsts/src/rules/package.json`

File: packages/jsts/src/rules/S5860/rule.ts [hunk 2]
```diff
@@ -326,7 +342,11 @@ function makeGroupKnowledge(
   index: number,
 ): GroupKnowledge {
   const name = node.name!;
-  const used = backreferences.some(backreference => backreference.resolved === node);
+  const used = backreferences.some(backreference =>
+    backreference.ambiguous
```
Comment (vdiez) [inline packages/jsts/src/rules/S5860/rule.ts:346]: shouldn't we use same helper here?

**Review by vdiez — APPROVED** [review]
just small comment

File: its/ruling/src/test/expected/jsts/ant-design/javascript-S6790.json [hunk 3]
```diff
@@ -1,6 +1,5 @@
 {
 "ant-design:components/upload/__tests__/upload.test.js": [
-37,
```
Comment (zglicz) [inline its/ruling/src/test/expected/jsts/ant-design/javascript-S6790.json:3]: This stopped showing up due to an update in `eslint-react-plugin` in v7.36 - commit https://github.com/jsx-eslint/eslint-plugin-react/pull/3807

File: its/ruling/src/test/expected/jsts/console/typescript-S6790.json [hunk 4]
```diff
@@ -6,51 +6,38 @@
...
-28,
 38
 ],
 "console:src/components/Help/Help.tsx": [
-28,
 36
 ],
 "console:src/components/PopupWrapper/PopupWrapper.tsx": [
-47,
-56
```
Comment (zglicz) [inline its/ruling/src/test/expected/jsts/console/typescript-S6790.json:18]: Same explanation for this file as above

File: its/ruling/src/test/expected/jsts/desktop/typescript-S6819.json [hunk 5]
```diff
@@ -1,4 +1,7 @@
 {
+"desktop:app/src/ui/app-menu/app-menu-bar-button.tsx": [
+190
+],
```
Comment (zglicz) [inline its/ruling/src/test/expected/jsts/desktop/typescript-S6819.json:4]: This seems to be a false positive stemming from using React components and us not understanding it. This was first reported here - https://community.sonarsource.com/t/s6819-prefer-tag-over-role-is-misleading/107348/7 and we have a jira task to track these - https://github.com/SonarSource/SonarJS/issues/4647

I'm not sure why it just now starting showing up in the ruling, but I think it's fair.

File: its/ruling/src/test/expected/jsts/moose/typescript-S1082.json [hunk 6]
```diff
@@ -2,6 +2,12 @@
 "moose:renderer/components/Cast/Cast.tsx": [
 51
 ],
+"moose:renderer/components/Downloads/Downloads.tsx": [
+130
+],
+"moose:renderer/components/FilesList/FilesList.tsx": [
+165
+],
```
Comment (zglicz) [inline its/ruling/src/test/expected/jsts/moose/typescript-S1082.json:10]: This is a true-positive, that wasn't triggering before
image

Comment (zglicz) [inline its/ruling/src/test/expected/jsts/moose/typescript-S1082.json:10]: True-positive

image

File: its/ruling/src/test/expected/jsts/redux/javascript-S6790.json [hunk 7]
```diff
@@ -1,11 +1,8 @@
 {
 "redux:examples/real-world/components/Explore.js": [
-19,
-26,
```
Comment (zglicz) [inline its/ruling/src/test/expected/jsts/redux/javascript-S6790.json:4]: same explanation as above

**Review by zglicz — COMMENTED** [review]
All of the ruling changes are accounted for.

**Pull request merged**